examples/posenet/utils.js

```diff
@@ -0,0 +1,229 @@
+                [0.75, mobileNet75Architecture],
+                [1.0, mobileNet100Architecture],
+                [1.01, mobileNet100Architecture],
+            ]);
+            this.tfmodel = ModelArch.get(Number(this._version));
+        }
+        this.model = new PoseNet(this.tfmodel, backend, Number(this._version), 
+                        Number(this._outputStride), input_size, this._type);
+        result = await this.model.createCompiledModel();
+        console.log('compilation result: ${result}');
+        this.initialized = true;
+    }
+
+    async predict(){
+        if(!this.initialized){
+            return;
+        }
+        let x = await getInput(this._inputElement);
+        this.canvasContext_single.clearRect(0, 0, this.canvasElement_single.width, this.canvasElement_single.height);
+        this.canvasContext_multi.clearRect(0, 0, this.canvasElement_multi.width, this.canvasElement_multi.height);
+        await loadImage(x, this.canvasContext_single);
+        await loadImage(x, this.canvasContext_multi);
+        let imageSize = [input_size[1], input_size[2], input_size[3]];
+        prepareInputTensor(this.inputTensor,this.canvasElement_multi, this._outputStride, imageSize);
+        let result = await this.model.compute_multi(this.inputTensor, this.heatmapTensor, 
+                this.offsetTensor, this.displacement_fwd, this.displacement_bwd);
+        let poses_multi = decodeMultiPose(this.heatmapTensor, this.offsetTensor, 
+                        this.displacement_fwd, this.displacement_bwd, 
+                        this._outputStride, this._maxDetection, this._minScore, 
+                        this._nmsRadius, toHeatmapsize(imageSize, this._outputStride));
+        let poses_single = decodeSinglepose(this.heatmapTensor, this.offsetTensor, 
+                            toHeatmapsize(imageSize, this._outputStride), this._outputStride);
+        poses_single.forEach((pose)=>{
+            if(pose.score>= this._minScore, this.canvasContext_single){
+                drawKeypoints(pose.keypoints, this._minScore, this.canvasContext_single);
+                drawSkeleton(pose.keypoints, this._minScore, this.canvasContext_single);
+            }
+        });
+
+        poses_multi.forEach((pose)=>{
+            if(pose.score>= this._minScore){
+                drawKeypoints(pose.keypoints, this._minScore, this.canvasContext_multi);
+                drawSkeleton(pose.keypoints, this._minScore, this.canvasContext_multi);
+            }
+        });
+    }
+
+    async loadmanifest(url){
+        let address = url+"manifest.json";
+        return fetch(address)
+        .then(function(response) {
+            return response.json();
+        })
+        .then(function(myJson) {
+            let data = {};
+            for(var i in myJson){
+                data[i] = myJson[i];
+            } 
+            return data;
+        });
+    }
+
+    async getvariable(url, binary){
```

Comment:
`getVariable`

Reply:
BTW, what's the reason for the function name `getVariable`? Its function looks like fetching file.